sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/xml/XmlInferSchema.scala

```diff
@@ -219,11 +219,27 @@ private[sql] object XmlInferSchema {
           dataTypes += inferredType
           nameToDataType += (field -> dataTypes)
 
+        case c: Characters if !c.isWhiteSpace =>
+          // This can be an attribute-only object
+          val valueTagType = inferFrom(c.getData, options)
+          nameToDataType += options.valueTag -> ArrayBuffer(valueTagType)
```

Comment:
IMHO it will make the loop much more complicated as we need to examine the case you mentioned above 
```
<ROW>4<tag>5</tag></ROW>
<ROW><tag>6</tag>7</ROW>
```